src/directives/on.js

```diff
@@ -24,9 +24,12 @@ export function registerListener(component, el, event, modifiers, expression, ex
         const listenerTarget = modifiers.includes('window')
             ? window : (modifiers.includes('document') ? document : el)
 
+        const useDebounce = modifiers.includes('debounce')
+        const wait = modifiers.length > 1 ? parseInt(modifiers[1]) : 250
+
         const handler = e => {
             if (isKeyEvent(event)) {
-                if (isListeningForASpecificKeyThatHasntBeenPressed(e, modifiers)) {
+                if (!useDebounce && isListeningForASpecificKeyThatHasntBeenPressed(e, modifiers)) {
```

Comment:
I'm not sure this line needs changing. We always want to skip a key event if the pressed key doesn't match what we defined whether or not we added the 'debounce' modifier.

For example:
```html
        <div x-data="{foo: 0}">
            <input x-on:keydown.a.debounce="foo = foo + 1" />
            <span x-text="foo"></span>
        </div>
```

When we press any keys, the counter will increment but it should not since we only want to increment it when we type `a`.

Reply:
@SimoTod Thanks for your feedback! You are right. For some reason while working on this, that was the only way I was able to make it work, but checked again and it works without that, so I removed it.

Reply:
You probably added it because, without that check, when you add 'debounce' the handler is always going to return without executing the code since `isListeningForASpecificKeyThatHasntBeenPressed` returns true. 

I think you need to handle that case inside `isListeningForASpecificKeyThatHasntBeenPressed`.

👍

Reply:
I am not sure now because I removed the check and both the automated test and the codepen with the updated code are working. Can you check here? https://codepen.io/jreviews/pen/XWJLoLZ

Reply:
Sorry, I typed the previous message quickly and I didn't explain myself really well.
The implementation is good but there is still an edge case which is not fully covered.

For example, check out this pen: https://codepen.io/SimoTod/pen/JjoQxXw
The first input doesn't have the debounce modifier, the second one does.
Both or them are supposed to increase the counter when the key 'a' is pressed (but they should not increase the counter when another key is pressed).

At the moment, the debounced one does not increment the counter. With the previous implementation, the counter was updated with any key.

I hope it makes sense.

Reply:
Ah yes! I remember now. The issue was with the regular inputs, no the debounced ones. Your demo is so much better. I'll have another go at it. Thanks.

Reply:
Can you check again? the issue with the key modifier should be fixed and I updated the tests.